packages/core/src/sync-realtime/index.ts

```diff
@@ -779,12 +780,23 @@ export const createRealtimeSync = (
     // Transaction Receipts
     ////////
 
-    const transactionReceipts = await Promise.all(
-      block.transactions
-        .filter(({ hash }) => requiredTransactionReceipts.has(hash))
-        .map(({ hash }) =>
-          _eth_getTransactionReceipt(args.requestQueue, { hash }),
-        ),
+    const blockReceipts = await _eth_getBlockReceipts(args.requestQueue, {
+      blockNumber: block.number,
+    });
+    
+    // Validate that block transaction receipts include all required transactions
+    const blockReceiptsTransactionHashes = new Set(
+      blockReceipts.map((r) => r.transactionHash),
+    );
+    for (const hash of Array.from(requiredTransactions)) {
+      if (blockReceiptsTransactionHashes.has(hash) === false) {
+        throw new Error(
+          `Detected inconsistent RPC responses. Transaction receipt with transactionHash ${hash} is missing in \`blockReceipts\`.`,
+        );
+      }
+    }
```

Comment:
Nice check

Reply:
What do you think of the checks in sync-historical?